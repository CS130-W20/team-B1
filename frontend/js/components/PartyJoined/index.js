import React, { Component } from 'react';
import { Button } from 'reactstrap';
import { Form, FormControl } from 'react-bootstrap'  
import SpotifyWebPlayer, { STATUS } from 'react-spotify-web-playback';

import SearchModal from '../SearchModal';

import '../css/SpotifyPlayer.css';

class PartyJoined extends Component {
	constructor(props) {
		super(props);
		this.state = {
			error: null,
			showModal: false,
			songOffset: 0,
			position: 0,
			autoPlay: true,
		}
	}

	handleCallback({ status, errorType, position, previousTracks }) {
		console.log(`~~~~~~~~~`)
		console.log(`previousTracks:`)
		console.log(previousTracks)
		console.log(`this.state.songOffset: ${this.state.songOffset}`);
		console.log(`position: ${position}`);
		console.log(`this.state.position: ${this.state.position}`);
		console.log(`~~~~~~~~~`)
		if (status === STATUS.ERROR && errorType === 'authentication_error') {
			localStorage.removeItem('token');
			// TODO: update token
			setToken('');
		}
		if (this.state.position < position && position >= 0) {
			this.setState({position: position});
		}
		if (previousTracks.length > this.state.songOffset || previousTracks.length == this.state.songOffset) {
			this.setState({songOffset: previousTracks.length});
		}
	}

	handleModal() {
		this.setState({showModal: !this.state.showModal});
	}

	handleSongAdd(data) {
		this.setState({showModal: false, autoPlay: false}); 
		this.props.handleSongAdd(data);
	}
	
	renderQueueItems(props) {
  		return (
  			<div className="list-group">
				<a href="#" className="list-group-item list-group-item-action listQueue">
					<div className="column left">
						<img className="albumArtwork" src={this.props.queue.albumArtwork} alt="cover" />
					</div>

					<div className="column middle">src={this.props.queue.songName} </div>

					<div className="column right buttonCard">
						<Button className="SkipButton" color="danger" onclick="skipSong()"> Skip </Button>
					</div>
				</a>
			</div>
  		);
	}

	render() {
		console.log(`songList: ${this.props.songList}`)
  	return (
			<div className="PartyJoined">
				<div className="title">
					<h1>{`${this.props.partyName} [Code: ${this.props.partyCode}]`}</h1>
				</div>

				<Button 
					variant="primary" 
					onClick={this.handleModal.bind(this)} 
					color="success"
					disabled={this.state.showModal}
				>
					Add Song
				</Button>

				<SearchModal 
					show={this.state.showModal}
					handleClose={this.handleModal.bind(this)}
					handleSongAdd={this.handleSongAdd.bind(this)}
				/>

				<div className="title">
					<h1>Song Queue </h1>
				</div>

				{this.props.error ? <small className="error">{this.props.error}</small> : null}

				{
					this.props.hosting ?
					<SpotifyWebPlayer
						autoPlay={this.state.autoPlay}
						callback={this.handleCallback.bind(this)}
						persistDeviceSelection
						play={false}
						magnifySliderOnHover
						showSaveIcon
						token={localStorage.getItem('token')}
						styles={{
							sliderColor: '#1cb954',
						}}
						name={'Director Web Player'}
						offset={this.state.songOffset}
						uris={this.props.songList}
						position={this.state.position}
					/>
					:
					null
				}
			</div>
		);
	}
}

export default PartyJoined;
