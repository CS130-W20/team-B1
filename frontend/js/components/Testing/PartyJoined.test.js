import React from 'react';
import renderer from 'react-test-renderer';

import PartyJoined from '../PartyJoined/index';

describe ('Testing Spotify Login Page', () => {
    it('correctly renders', () => {
        const wrapper = renderer.create(<PartyJoined />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});