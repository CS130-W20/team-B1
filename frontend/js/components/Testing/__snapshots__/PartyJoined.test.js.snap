// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Spotify Login Page correctly renders 1`] = `
<div
  className="PartyJoined"
>
  <div
    class="searchDiv"
  >
    <form
      class="searchBar"
      className=""
    >
      <input
        className="mr-sm-2 form-control"
        placeholder="Search"
        type="text"
      />
    </form>
  </div>
  <div
    class="title"
  >
    <h1>
      Song Queue 
    </h1>
  </div>
  <div
    className="listViewSongQueue"
  >
    <div
      class="list-group"
    >
      <a
        class="list-group-item list-group-item-action listQueue"
        href="#"
      >
        <div
          class="column left"
        >
          <img
            alt="W3Schools.com"
            class="albumArtwork"
            src="https://www.w3schools.com/images/w3schools_green.jpg"
          />
        </div>
        <div
          class="column middle"
        >
          Name of Song
        </div>
        <div
          class="column right buttonCard"
        >
          <button
            aria-label={null}
            className="SkipButton btn btn-danger"
            onClick={[Function]}
          >
            Skip
          </button>
        </div>
      </a>
    </div>
  </div>
</div>
`;
